Include/internal/pycore_bitutils.h

```diff
@@ -82,6 +84,45 @@ _Py_bswap64(uint64_t word)
 }
 
 
+// Population count: count the number of 1's in 'u'
+// (number of bits set to 1), also known as the hamming weight.
+//
+// Implementation note. CPUID is not used, to test if x86 POPCNT instruction
+// can be used, to keep the implementation simple. For example, Visual Studio
+// __popcnt() is not used this reason. The clang and GCC builtin function can
+// use the x86 POPCNT instruction if the target architecture has SSE4a or
+// newer.
+static inline int
+_Py_popcount32(uint32_t x)
+{
+#if (defined(__clang__) || defined(__GNUC__)) && SIZEOF_INT == 4
+    return __builtin_popcount(x);
+#elif (defined(__clang__) || defined(__GNUC__)) && SIZEOF_INT == 8
+    return __builtin_popcountl(x);
+#else
+    // 32-bit SWAR (SIMD Within A Register) popcount
+
+    // Binary: 0 1 0 1 ...
+    const uint32_t M1 = UINT32_C(0x55555555);
```

Comment:
I don't think the `UINT32_C` macro serves any purpose here; it could be dropped.